openpype/hosts/houdini/api/lib.py

```diff
@@ -1054,3 +1056,71 @@ def add_self_publish_button(node):
     template = node.parmTemplateGroup()
     template.insertBefore((0,), button_parm)
     node.setParmTemplateGroup(template)
+
+class Context:
+    main_window = None
+    context_action_item = None
+    project_name = os.getenv("AVALON_PROJECT")
+    # Workfile related code
+    workfiles_launched = False
+    workfiles_tool_timer = None
+
+
+def launch_workfiles_app():
+    """Show workfiles tool on nuke launch.
+
+    Trigger to show workfiles tool on application launch. Can be executed only
+    once all other calls are ignored.
+
+    Workfiles tool show is deferred after application initialization using
+    QTimer.
+    """
+
+    if Context.workfiles_launched:
+        return
+
+    Context.workfiles_launched = True
+
+    # get all imortant settings
+    open_at_start = env_value_to_bool(
+        env_key="OPENPYPE_WORKFILE_TOOL_ON_START",
+        default=None)
+
+    # return if none is defined
+    if not open_at_start:
+        return
+
+    # Show workfiles tool using timer
+    # - this will be probably triggered during initialization in that case
+    #   the application is not be able to show uis so it must be
+    #   deferred using timer
+    # - timer should be processed when initialization ends
+    #       When applications starts to process events.
+    timer = QtCore.QTimer()
+    timer.timeout.connect(_launch_workfile_app)
+    timer.setInterval(100)
+    Context.workfiles_tool_timer = timer
+    timer.start()
```

Comment:
Is this really needed? Doesn't it just work solely due to the `hdefereval` call?

I'd expect something like this to be the full implementation:
```python
def launch_workfiles_app():
    if os.environ["AVALON_LAST_WORKFILE"]:
        # Host launched with last workfile to be opened
        # as such we do not show the workfile tool
        log.debug(
            "Last workfile path found so workfile tool won't be launched."
        )
        return
    
    # Parent tool to main Houdini window - if not found, we force the
    # tool to be on top to avoid it remaining hidden behind e.g. Houdini
    # window.
    # TODO: Check whether there are any cases where Houdini's main window
    #   does not exist yet.
    parent = get_main_window()
    on_top = not parent
    host_tools.show_workfiles(parent=parent, on_top=on_top)